process/sync/baseSync.go

```diff
@@ -372,6 +366,16 @@ func (boot *baseBootstrap) receivedHeaderNonce(nonce uint64, shardId uint32, has
 		nonce,
 		core.ToB64(hash)))
 
+	header, err := boot.getHeaderFromPool(hash)
```

Comment:
might extract this into a func in which you can easily drop the else branch

Reply:
done